src/reference/90-Developers-Guide/02-Coding-Guideline/00.md

```diff
@@ -90,83 +90,87 @@ Use [MiMa][mima].
 
 #### Public traits should contain `def` declarations only
 
+TODO: "results to codegen at subclass" is unclear
```

Comment:
For the full explanation you should watch Josh's talk https://skillsmatter.com/skillscasts/3269-binary-resilience.
Both "causes" and "results in" suggestions are fine, I think but the larger point I am trying to convey is that `val` in trait reaches into the subclass code generation. This is undesirable for growable API because when we change the trait, it will require plugin authors to recompile their code against ours or else their plugin would blow up during runtime.